### PR TITLE
Tests exhibiting performance problems with floating point equalities

### PR DESCRIPTION
**Pull request by peterschrammel**
See #4337.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/cbmc/Float-equality1/main.c [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include <assert.h>
+
+void main() {
+  double a, b, c;
+  __CPROVER_assume(a + b > c);
+#ifdef EQUALITY
+  double x = a, y = b, z = c;
+  assert(!(z > x + y) );
+#else 
+  assert(!(c > a + b) );
```
Comment (tautschnig) [inline regression/cbmc/Float-equality1/main.c]: This test isn't clang-format clean. (Opening `{` on new line, no space before `)` (twice).)

File: regression/cbmc/Float-equality1/test_no_equality.desc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CORE
```
Comment (tautschnig) [inline regression/cbmc/Float-equality1/test_no_equality.desc]: The SMT-backend is broken, and this test again shows this. Add `broken-smt-backend`. Preferably also to the `KNOWNBUG` test.

**Review by tautschnig — APPROVED** [review]
This is failing CI as explained below, but otherwise ready to be merged.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b3c9adec0a634f61ed5c5e35fd104c14f525531e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103505025

**Pull request merged**